lib/context_adapter_communication.js

```diff
@@ -518,6 +534,22 @@ function getThirdPartyPayload(serviceDescriptor, operationDescriptor) {
 }
 
 /**
+ * Encodes the response sent to the Context Broker so no forbidden characters are used
+ *  The forbidden characters not allowed by the Context Broker are: < > " ' = ; ( )
+ * @param str The string to be encoded
+ * @return {string} The encoded string
+ */
+function encodeResponse(str) {
+  // encodeURIComponent() escapes all characters except the following: alphabetic, decimal digits, - _ . ! ~ * ' ( )
+  var encodedStr = encodeURIComponent(str);
```

Comment:
I'm not sure it's a good idea to URL encode the response you are going to use to fill a JSON that no one expects to be URL Encoded... You will end up with an unreadable messy text full of control characters in a log or in a text box, as no one will decode it.